src/Components/test/testassets/Ignitor/BlazorClient.cs

```diff
@@ -160,115 +170,162 @@ public Task SelectAsync(string elementId, string value)
             return ExpectRenderBatch(() => elementNode.SelectAsync(HubConnection, value));
         }
 
-        public async Task ExpectRenderBatch(Func<Task> action, TimeSpan? timeout = null)
+        public async Task<CapturedRenderBatch> ExpectRenderBatch(Func<Task> action, TimeSpan? timeout = null)
         {
             var task = WaitForRenderBatch(timeout);
             await action();
-            await task;
+            return await task;
         }
 
-        public async Task ExpectJSInterop(Func<Task> action, TimeSpan? timeout = null)
+        public async Task<CapturedJSInteropCall> ExpectJSInterop(Func<Task> action, TimeSpan? timeout = null)
         {
             var task = WaitForJSInterop(timeout);
             await action();
-            await task;
+            return await task;
         }
 
-        public async Task ExpectDotNetInterop(Func<Task> action, TimeSpan? timeout = null)
+        public async Task<string> ExpectDotNetInterop(Func<Task> action, TimeSpan? timeout = null)
         {
             var task = WaitForDotNetInterop(timeout);
             await action();
-            await task;
+            return await task;
         }
 
-        public async Task ExpectCircuitError(Func<Task> action, TimeSpan? timeout = null)
+        public async Task<string> ExpectCircuitError(Func<Task> action, TimeSpan? timeout = null)
         {
             var task = WaitForCircuitError(timeout);
             await action();
-            await task;
+            return await task;
+        }
+
+        public async Task<Exception> ExpectDisconnect(Func<Task> action, TimeSpan? timeout = null)
+        {
+            var task = WaitForDisconnect(timeout);
+            await action();
+            return await task;
```

Comment:
Should this be return `Task.WhenAny(task, action)`? You don't want to await action because if action never finishes you never timeout.

Reply:
This is consistent with all of the others here, and I don't feel obliged to change it. That would also roll back the change to return the result from these methods.

Reply:
Doesn't that mean it is consistently **wrong** then?

Reply:
No, I would say what's wrong would be passing in code via `action` to this method that runs indefinitely. This hasn't been a problem in practice, and if that were the case I'd prefer to address that instead.